datacube/index/abstract.py

```diff
@@ -1018,16 +1018,53 @@ def __init__(self, index):
     @abstractmethod
     def get(self,
             id_: DSID,
-            include_sources: bool = False
-           ) -> Optional[Dataset]:
+            include_sources: bool = False,
+            include_deriveds: bool = False,
+            max_depth: int = 0
+            ) -> Optional[Dataset]:
         """
         Get dataset by id
 
+        - Index drivers supporting the legacy lineage API:
+
         :param id_: id of the dataset to retrieve
-        :param include_sources: get the full provenance graph?
+        :param include_sources: include the full provenance tree of the dataset.
+
+
+        - Index drivers supporting the external lineage API:
+
+        :param id_: id of the dataset to retrieve
+        :param include_sources: include the full provenance tree for the dataset.
+        :param include_deriveds: include the full derivative tree for the dataset.
+        :param max_depth: The maximum depth of the source and/or derived tree.  Defaults to 0, meaning no limit.
         :rtype: Dataset model (None if not found)
         """
 
+    def _check_get_legacy(self,
+                          include_deriveds: bool = False,
+                          max_depth: int = 0
+                          ) -> None:
+        """
+        Index drivers implementing the legacy lineage API can call this method to check get arguments
+        """
+        if not self._index.supports_external_lineage:
+            if include_deriveds:
+                raise NotImplementedError(
+                    "This index driver only supports the legacy lineage data - include_deriveds not supported."
+                )
+            if not self._index.supports_external_lineage and (include_deriveds or max_depth > 0):
+                raise NotImplementedError(
+                    "This index driver only supports the legacy lineage data - max_depth not supported."
+                )
+
+    def _get_legacy(self, id_: DSID, include_sources: bool = False) -> Optional[Dataset]:
```

Comment:
Doesn't seem to be used

Reply:
Yes, thanks.  Relic of a first attempt.  Will clean up.